gradio/external.py

```diff
@@ -429,6 +460,10 @@ def streamline_config(config: dict) -> dict:
     config = streamline_config(config)
     api_url = "https://hf.space/embed/{}/api/predict/".format(model_name)
     headers = {"Content-Type": "application/json"}
+    ws_url = "wss://spaces.huggingface.tech/{}/queue/join".format(model_name)
```

Comment:
I didn't know this was the URL? Why is it on a completely different domain than the `api_url`?

Reply:
Not sure. My guess is that the spaces team restricts the ports/paths with which you can connect to the underlying machine.